File: fortran/knagpal-COE322-inclass5.f90
       !Fortran fifth in-Class Lab: Prints out all the sum of squares
       !(u^2 + v^2 = w^2) in the range 1-100, w/o duplicates
       !Kartik Nagpal - Kn8767
       
       program SumSquares

       implicit none

       integer :: u, v
       real(4) :: d


       do u = 1, 100
           do v = 1, 100
                d = sqrt(Real(u**2 + v**2))
                if(d == int(d) .and. u < v .and. d < 100) then
                    print *, u, "^2 + ", v, "^2 = ", d, "^2"
                end if
           end do
       end do

       end program sumSquares

File: C++/knagpal-COE322-inclass5.cpp
//Program Name:	Sum of Squares - prints first 100 sums 
//Created by Kartik Nagpal - kn8767 - kartiknagpal@utexas.edu

#include <iostream>
#include <cmath>
using std::cin;
using std::cout;
using std::endl;
using std::string;

void sum(int &n, int &S, int &m);

int main() {
	int n = 1;
	int S0 = 1;
	int max = 100;
	sum(n, S0, max);
	return 0;
}

void sum(int &n, int &S, int &m) //n = current count; S = sum; m = max;
{
	cout << "S(" << n << ")= " << S << endl;
	if(n < m)
	{
		S += pow(n, 2);
		n++;
		sum(n, S, m);
	}
}
